packages/components/src/internal/util/utils.ts

```diff
@@ -447,6 +447,25 @@ export function isNonNegativeFloat(value: number | string): boolean {
     return isFloat(value) && Number(value) >= 0;
 }
 
+export function isAllowedSampleAmount(value: unknown): boolean {
+    if (!value) return true;
+
+    if (typeof value === 'number' || typeof value === 'string')
+        return isNonNegativeFloat(value) && Number.isFinite(Number(value));
+
+    return false;
+}
+
+export const isValidSampleAmountWithError = (v: any): any => {
```

Comment:
I don't love the name of this method. Since it's really just returning an error message for invalid amounts, perhaps `getInvalidSampleAmountMessage` would be appropriate.